modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/services/MetagraphMessageService.scala

```diff
@@ -0,0 +1,62 @@
+package org.tessellation.node.shared.infrastructure.snapshot.services
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.ValidatedNec
+import cats.effect.Async
+import cats.syntax.functor._
+import cats.syntax.validated._
+
+import org.tessellation.node.shared.domain.seedlist.SeedlistEntry
+import org.tessellation.node.shared.infrastructure.snapshot.services.MetagraphMessageService.MetagraphMessageServiceErrorOr
+import org.tessellation.node.shared.infrastructure.snapshot.storage.MetagraphMessageStorage
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.metagraphMessage.{AdminFeeKeyMessage, StakingMessage}
+import org.tessellation.security.signature.SignedValidator
+import org.tessellation.security.signature.SignedValidator.{SignedValidationError, SignedValidationErrorOr}
+import org.tessellation.security.{Hasher, SecurityProvider}
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+
+trait MetagraphMessageService[F[_]] {
+  def offerAdminFeeKeyMessage(message: AdminFeeKeyMessage): F[MetagraphMessageServiceErrorOr[Unit]]
+  def offerStakingMessage(message: StakingMessage): F[MetagraphMessageServiceErrorOr[Unit]]
+}
+
+object MetagraphMessageService {
+
+  def make[F[_]: Async: SecurityProvider: Hasher](
+    validator: SignedValidator[F],
+    seedlist: Option[Set[SeedlistEntry]],
+    storage: MetagraphMessageStorage[F]
+  ): MetagraphMessageService[F] =
+    new MetagraphMessageService[F] {
+      val peers = seedlist.map(_.map(_.peerId))
+
+      def offerAdminFeeKeyMessage(message: AdminFeeKeyMessage): F[MetagraphMessageServiceErrorOr[Unit]] =
```

Comment:
just `offer(message: AdminFeeKeyMessage)` and `offer(message: StakingMessage)` would be enough IMO. the parameter would be self explaining and you would have just `messageService.offer(message)`

Reply:
changed to single `offer` method